hivemind/averaging/matchmaking.py

```diff
@@ -160,8 +158,6 @@ async def _request_join_potential_leaders(self, timeout: Optional[float]) -> Gro
                         elif len(self.current_followers) > 0:
                             await self.leader_disband_group()
                         continue
-                except (concurrent.futures.CancelledError, asyncio.CancelledError):
```

Comment:
The reason we needed this was to avoid propagating CancelledError that would trigger from networking issues.

It would appear that this is indeed no longer necessary, but we may need to revert this one specific change if we encounter stochastic collaboration-wide freezes after peers disconnect.

Reply:
I've reverted this change where I'm not sure, see the [comment](https://github.com/learning-at-home/hivemind/pull/357#discussion_r695271601).